frameworks/shared/caller.py

```diff
@@ -93,6 +94,12 @@ def run_in_venv(caller_file, script_file: str, *args,
     ser_config = options['serialization']
     env = options['env'] or ns()
 
+    # To honour any arguments about tmpdir, we copy over
+    # the TMP, TEMP and TMPDIR environment variables
+    if retain_tmp_env:
+        for env_var in ["TMP", "TEMP", "TMPDIR"]:
+            env[env_var] = os.environ.get(env_var)
```

Comment:
adapt according to suggestion above.

Also, having an env variable with `None` value is not exactly the same as having no env variable, so I'd rather suggest to use:
```python
if retained_env_vars:
    for env_var in retained_env_vars:
        env_val = os.environ.get(env_var)
        if env_val is not None:
            env[env_var] = env_val
```